### ISSUE TITLE
custom element doesn't exist: teamtracker-card

### ISSUE THREAD

**Issue by tomofdarkness** [vasqued2/ha-teamtracker-card#87]
I get this error when I load a dashboard with the card in Chrome on two computers, and I briefly see it before the card loads in Safari on one of those before it loads.  Card loads just fine in the companion app on iOS, MacOS, and Android.  Attached are screenshots in iOS companion app and in Chrome.

![image](https://github.com/vasqued2/ha-teamtracker-card/assets/6440018/7f80de69-c3b7-4382-8bdf-982b65d8b287)

![image](https://github.com/vasqued2/ha-teamtracker-card/assets/6440018/d3931733-0f30-48d0-8ecd-be2eb3e39675)

**Comment by vasqued2** [comment]
First try clearing the cache on the browsers that it isn't working on and the refresh the page.  Did you install using HACS?

**Comment by tomofdarkness** [comment]
I did clear the cache and refresh, one of the browsers had never logged in
before, and I did install via hacs. Thanks!

**Comment by vasqued2** [comment]
Try redownloading the card (option in the three dots menu in the top right) in HACS using the browser with the problem and then select Reload when given the option.

**Comment by vasqued2** [comment]
Assumed resolved.

**Comment by boggsdabeast** [comment]
@vasqued2 I get this same problem when casting to google nest hub. The dashboard works just fine on mobile/pc. 
I did try redownload card in hacs but obviously I'm casting so it wouldn't be the device having the problem. 
Don't think it's possible to clear cache on nest hub? I'm not finding anything in settings. Hope I don't have to factory reset since I have multiple hubs all having this problem. It used to work previously then few weeks ago it stopped working.

Ideas for this?
![image](https://github.com/user-attachments/assets/76d1c7af-229f-4f20-b9dd-eb84f89345fc)

**Comment by vasqued2** [comment]
I'm not familiar with the Nest Hub.  Did you do an upgrade on HA, the card, or the sensor that caused it to break, or did it just stop working?

**Comment by boggsdabeast** [comment]
to be honest, I don't recall how it initially broke or if I had done anything before I noticed the issue. It has been a few weeks and I was slowly looking for solutions but am lazy and was hoping a fix would just magically come back lol. I may have done an update of the team tracker integration then noticed the issue. I then updated HA Core/OS hoping that would help but it did not. Restarted VM, host and nest device with no luck either.

**Comment by vasqued2** [comment]
You can go into HACS and redownload old versions.

I added URL hotspots in v0.14.x, you might try to got back and redownload v0.13.x and see if that corrects it.

Also, the HA 2024.8.0 release introduced a change that caused a lot more devices to be classified as "legacy" which caused them to use a different HTML component.  This has also caused issues on some devices.  Although they appear to be rendering issues, it's at least considering that as a possibility too.  What HA version are you on?

**Comment by boggsdabeast** [comment]
I am currently on HA Core 2024.8.2 Supervisor 2024.08.0 OS 13.0 Frontend 20240809.0 
I am on team tracker card v0.14.1 and downgraded to v0.13.1 and still have the issue :(

I do have two backups since Aug1 that should be on previous HAOS/Core versions so later this week I can try and restore them to see if problem exists there as well.

**Comment by vasqued2** [comment]
Do you know of any way to see the output on the console/browser console and see if there are any errors?

I just released v0.14.2-beta1 to fix the iPad problem.  You can download it from HACS if you have show beta checked.  I doubt if it will fix it but you could try that too.  If your device is now considered legacy, it might help.

**Comment by boggsdabeast** [comment]
I restored a backup which was on HA Core 2024.1.2 Supervisor 2024.08.0 OS 13.0 Frontend 20240104.0 
this backup was also on team tracker v0.13 and team tracker card v0.13 and still has the issue :(

I'm pretty sure when I took the backup there was no issue so it must be something internal to my system??? 

I think the only error I can see in logs related to chromecast is this:
```
Logger: frontend.js.modern.202408090
Source: components/system_log/__init__.py:300
First occurred: 8:48:47 AM (1 occurrences)
Last logged: 8:48:47 AM

Uncaught error from Chrome 122.0.0.0 on Chromecast 1.56.500000 SyntaxError: The requested module './const.js' does not provide an export named 'ERROR_HEADSHOT_URL' The requested module './const.js' does not provide an export named 'ERROR_HEADSHOT_URL'::
```

**Comment by vasqued2** [comment]
v0.13.1 of the teamtracker card introduced the ability to display athlete headshots.  It looks like this is a caching issue where you have v0.13.1 or later of some files but v0.13.0 or earlier of the `const.js` file.

Clearing the cache so all the files download again would be the ideal way to solve it but since you don't know how to clear the cache I'd try 2 things.  

You can first try reverting to v0.13.0 of the card or earlier and see if that get you back to the earlier version of the card that will match the const.js file.  You can use `debug:true` in the card configuration to sure the card version at the top to confirm the version you are on.

You could also try replacing `ERROR_HEADSHOT_URL` with `"https://cdn-icons-png.freepik.com/512/9706/9706583.png"` (you need the double quotes" in the file `set_defaults.js`.  If you edit this file, you will have to delete the zip files otherwise it won't pick up your changes.

I will also look for a way to programmatically force all files to be downloaded but not sure if that is possible.

**Comment by vasqued2** [comment]
Try adding `/hacsfiles/ha-teamtracker-card/const.js?hacstag=52640868201420` to your resources and see if that forces the `const.js` file to be redownloaded.  Definitely tell me if this works.

**Comment by chiefspecialk** [comment]
0.14.2 and 0.14.2-beta2 both gave me the error from original post. I rolled back to 0.14.2-beta1 and works fine.

**Comment by fcortijos** [comment]
I rolled back to 0.14.1 and works fine.

**Comment by TarheelGrad1998** [comment]
I also had this issue with 0.14.2.  Clearing cache resolved it in Chrome, but not in Arc (which is my primary browser now).  I tried adding const.js to resources as above and that didn't fix it either.  So I gave up and went back to 0.14.1 and all is well.  Strange.

**Comment by vasqued2** [comment]
Just downloaded Arc and 0.14.2 is working on mine.  But I was already on 0.14.2 and didn't upgrade.  Can you take a look at the console and tell me if there are any error messages displayed.

**Comment by vasqued2** [comment]
I reverted to an older version of teamtracker card and the messages in the console indicate that even after clearing the cache, some prior .js files remain.  This give me enough to run on.  I think I will be able to detect when files across multiple card versions are used, but it's going to take a while to figure out how to handle.

**Comment by vasqued2** [comment]
For the Arc browser, selecting 'Clear Cache' from the 'Site Control Center' at the top did NOT correct the problem.

I was able to get the card to work after an upgrade (or downgrade) with the following:

- Select 'All Site Settings' from the 'Site Control Center' at the top.
- Select 'Privacy and security'
- Pick 'Delete browsing data'
- Set the time range to 'All Time'
- Pick 'Cached images and files' from the popup.
- Select the 'Delete Data' button.

Refresh the page with Home Assistant on it and the card should display correctly.

Give how easy it is to replicate inconsistencies in cached files with the Arc browser, it will give me something to experiment with to detect and see if I can come up w/ a workaround.  I do not expect this to be quick, however.

If the above steps don't work w/ the Arc browser, let me know.

**Comment by TarheelGrad1998** [comment]
Fixed it for me.  Thanks, @vasqued2

**Comment by vasqued2** [comment]
UPDATE:  This issue is caused when some `teamtracker-card` files are updated with a new version while other older files remain unrefreshed in cache.  The short-term fix is to clear the cache of the impacted browser/app and refresh the impacted page.  This should be the first step to try.

If that does not work, check the console and see what/if any error messages are listed.  The version number of the card may be printed in the console.

Post the details of any errors listed in the console, the version number, and the browser/app you are having problems with in this thread.

The long-term fix is to use a JavaScript bundler to bundle all of the JavaScript files into a single file so they can't get out of sync.  I am researching this option but it will take a while.

**Comment by vasqued2** [comment]
`v014.3-beta1` should fix this issue going forward.

@boggsdabeast , this beta should fix it on your Nest Hub too.  Let me know if you are still have problems after trying it.

**Comment by fcortijos** [comment]
v014.3-beta1 is working fine. Tks

**Comment by gemmaker** [comment]
> `v014.3-beta1` should fix this issue going forward.
> 
> @boggsdabeast , this beta should fix it on your Nest Hub too. Let me know if you are still have problems after trying it.

V14.3-beta1 also fixed this for me (I use the HA android app). I had tried reloading and also roll back to previous versions which didn't work, but beta did. It wasn't the headshots either, as that was working for me previously. Ta.

**Comment by boggsdabeast** [comment]
@vasqued2 I had tried some stuff over the weekend just hadn't posted here, but everything I tried (removing card from HA, factory reset google hub, etc) was unsuccessful.

Downloaded v014.3-beta1 appears to have worked for me though!

**Comment by vasqued2** [comment]
Great to hear.  Thanks for trying and getting back to me.  

So far only positive feedback.  Unless I hear any issues soon, I'll have an official release later today or tomorrow.

**Comment by vasqued2** [comment]
Resolved w/ v0.14.3
